src/Dfc.CourseDirectory.Core/DataStore/Sql/Queries/UpsertCourses.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using Dfc.CourseDirectory.Core.Models;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.Core.DataStore.Sql.Queries
+{
+    public class UpsertCourses : ISqlQuery<None>
+    {
+        public IEnumerable<UpsertCoursesRecord> Records { get; set; }
+    }
+
+    public class UpsertCoursesRecord
+    {
+        public Guid CourseId { get; set; }
+        public CourseStatus CourseStatus { get; set; }
+        public DateTime CreatedOn { get; set; }
+        public string CreatedBy { get; set; }
+        public DateTime? UpdatedOn { get; set; }
+        public string UpdatedBy { get; set; }
+        public int? TribalCourseId { get; set; }
+        public string LearnAimRef { get; set; }
+        public int ProviderUkprn { get; set; }
+        public string CourseDescription { get; set; }
+        public string EntryRequirements { get; set; }
+        public string WhatYoullLearn { get; set; }
+        public string HowYoullLearn { get; set; }
+        public string WhatYoullNeed { get; set; }
+        public string HowYoullBeAssessed { get; set; }
+        public string WhereNext { get; set; }
+        public IEnumerable<UpsertCoursesRecordCourseRun> CourseRuns { get; set; }
+    }
+
+    public class UpsertCoursesRecordCourseRun
```

Comment:
I prefer class-per-file, gives me one less thing to think about

Reply:
This is consistent with other SQL-sync commands. Happy to review the approach but I'd rather keep consistent until then.

Reply:
Agreed